### ISSUE TITLE
[REVIEW]: Flint: a simulator for biological and physiological models in ordinary and stochastic differential equations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2331]
**Submitting author:** @tabe (<a href="http://orcid.org/0000-0002-7074-4561">Takeshi Abe</a>)
**Repository:** <a href="https://github.com/flintproject/Flint" target ="_blank">https://github.com/flintproject/Flint</a>
**Version:** Flint-2.4.0RC1-JOSS 
**Editor:** @majensen
**Reviewers:** @mstimberg, @dawbarton
**Archive:** <a href="https://doi.org/10.5281/zenodo.4017040" target="_blank">10.5281/zenodo.4017040</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5/status.svg)](https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5"><img src="https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5/status.svg)](https://joss.theoj.org/papers/ba63c7df89b7f27f5d8003bc42cdf5e5)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/flintproject/Flint" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mstimberg & @dawbarton, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @majensen know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/flintproject/Flint">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tabe) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/flintproject/Flint">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tabe) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @funasoul, @mstimberg it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by majensen** [comment]
@whedon remind @majensen in 5 days

**Comment by whedon** [comment]
Reminder set for @majensen in 5 days

**Comment by majensen** [comment]
@arfon - should I remind @whedon to remind me? 😄

**Comment by arfon** [comment]
:-) we'll see...

**Comment by mstimberg** [comment]
I've started reviewing but I have to say that I had some difficulties to get things going. I'm on Ubuntu, so none of the provided binary packages applies to my situation. I had a look into installing from source, but it seems to require many dependencies that are non-standard and would have to be installed from source as well. Since I did not want to go down that rabbit hole I installed CentOS 8 in a Virtual Machine and used the provided binaries instead. I can confirm that the installation works, but then again things get a bit difficult: the only concrete example file mentioned in the user documentation (I did not look at the Youtube videos, though)  is a file "distributed as part of the PhysioDesigner installation". PhysioDesigner seems to be a related project, but it is not mentioned in the JOSS article. The [Flint website](https://flintproject.github.io) has a link to www.physiodesigner.org, but this website seems to be down. Same for https://physiome.jp, the other link to an external source on the website (without any explanation what it refers to). I finally got the simulator running with an SMBL file from elsewhere and I will continue to do the review, but as first suggestions for improvement @tabe:
* Give some details about related projects such as PhysioDesigner or  physiome.jp in the JOSS article
* Provide clear instructions how to get example files in the documentation, or distribute them with the software itself
* The website has packages for CentOS/RHEL and the github repository also mentions them, but the user guide only has instructions for Windows/OS X

**Comment by tabe** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by tabe** [comment]
Dear @mstimberg,

> 
>     * Give some details about related projects such as PhysioDesigner or  physiome.jp in the JOSS article
I've mentioned the physiome.jp project and related tools in the last revision of paper.md.
Confirmed their sites are out of reach now, I'm not sure when they will come back though.

> 
>     * Provide clear instructions how to get example files in the documentation, or distribute them with the software itself
I'm going to reconsider the examples in the user guide, and include the model files in the distribution, too.

> 
>     * The website has packages for CentOS/RHEL and the github repository also mentions them, but the user guide only has instructions for Windows/OS X
I will update the user guide in order to include descriptions specific to CentOS/RHEL.
Thank you for your suggestions.

**Comment by mstimberg** [comment]
Thanks for the changes so far, I was now able to install Flint from source on Ubuntu 18.04. I opened two issues in the project's repository:
* Installation instructions: flintproject/Flint#15
* User manual: flintproject/Flint#16

Here a few general comments about the article, the software, and the website:

* The last sentence of the first paragraph of the introduction ("As a consequence, ...") is rather unclear  and would benefit from rewriting, I think.
* Still in the introduction, the authors states "aims to establish a computational platform for multiscale in silico studies on physiome". I have to admit that I have not come across the term "physiome" so far, so it might benefit from a little explanation/reference for the general audience.
* The main weakness of the introduction is that the unique contribution of Flint remains a bit unclear. Is it that it can not only simulate SBML models (as many other simulators), but also models on a higher level defined in PHML that include low-level descriptions in SBML? Does it offer benefits for users only interested in SBML models? 
* It seems a bit odd that the implementation section only focusses on the parameter fitting functionality. From what I understood, this is more of a "bonus feature" (which is not even mentioned in the user manual). Of course the paper is not meant to replace technical documentation, but maybe there is something to say about the high level functionality of the general software here (e.g. how it leverages existing libraries for its functionality, or how it uses multi-threading, etc.)? 
* The README mentions that the project's website contains "its user manual and tutorials", but the ["Tutorial videos on YouTube"](https://www.youtube.com/user/PhysioDesigner) linked on that site seem to be tutorials for `PhysioDesigner`, not for `Flint` (at least judging from their titles).

Minor issues:
* In the last sentence of the first paragraph, the abbrevations that are directly followed by citations (e.g. "(SBML) (Hucka et al., 2003)") look a bit ugly. I think you can use `[SBML; @hucka_systems_2003]` instead of `(SBML) [@hucka_systems_2003]` (did not test this, though).

**Comment by tabe** [comment]
Dear @mstimberg,

>     * Installation instructions: [flintproject/Flint#15](https://github.com/flintproject/Flint/issues/15)
> 
>     * User manual: [flintproject/Flint#16](https://github.com/flintproject/Flint/issues/16)
Thank you for opening the above issues, we've started working to solve them.

> Here a few general comments about the article, the software, and the website:
They are all reasonable, so revising our paper to amend the weakness you pointed out.

>     * The README mentions that the project's website contains "its user manual and tutorials", but the ["Tutorial videos on YouTube"](https://www.youtube.com/user/PhysioDesigner) linked on that site seem to be tutorials for `PhysioDesigner`, not for `Flint` (at least judging from their titles).
The videos do include how to simulate PHML models by Flint while the focus is how to edit them with PhysioDesigner.
We will prepare a new tutorial dedicated to Flint soon.

**Comment by tabe** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by mstimberg** [comment]
Dear @tabe: given that you left the issues open, I assume you are still working on them? Let me/us know when you want us to have another look.

I am afraid I will be out of office from July 13th to July 31st, though, so I cannot guarantee any timely response during that period.

**Comment by majensen** [comment]
@mstimberg, just want to thank you for your work so far. @funasoul -- have you had an opportunity to take a look at this software? Thanks very much -

**Comment by tabe** [comment]
@mstimberg,

> Dear @tabe: given that you left the issues open, I assume you are still working on them? Let me/us know when you want us to have another look.
> 
Yes, I will add a comment on each of the issues once it is done, to let you know.

**Comment by dawbarton** [comment]
> Hello @dawbarton - I know you are engaged in another JOSS review at the moment, but I wondered if [this work](https://github.com/openjournals/joss-reviews/issues/2331) would be interesting to you. We could use your assistance! Thanks for your consideration, and any suggestions for alternative reviewers. --@majensen

I'd be happy to take a look at this but realistically I won't get chance to take a look until w/c 8 August (or slightly after depending on what is happening in that week). If that's fine with you, I'll do it.

**Comment by majensen** [comment]
Thanks @dawbarton that would be great.

**Comment by majensen** [comment]
@whedon add @dawbarton as reviewer

**Comment by whedon** [comment]
OK, @dawbarton is now a reviewer

**Comment by majensen** [comment]
Added a checklist for you in the header @dawbarton - thanks again

**Comment by majensen** [comment]
Hi @dawbarton - just following up; are you still able to review as you anticipated? Thanks

**Comment by dawbarton** [comment]
> Hi @dawbarton - just following up; are you still able to review as you anticipated? Thanks

Yes, I can do it. Hope to take a look tomorrow or Friday; early next week at the latest.

**Comment by dawbarton** [comment]
Installation: I like the list of dependencies for different Linux distributions. The one dependency that isn't listed is gnuplot - it might be worth adding that to the list given that it is required for viewing the plots. I built the package from source under Linux using the supplied Makefile and it was a smooth process.

A statement of need: I can't see any sort of statement of need, comparison with other software, or intended audience in the documentation. There is a clear description of what the software does though. (And I can infer from that who the intended audience is, but it would be good to express it more clearly.) Simply taking some of the text from the JOSS paper and putting in an appropriate place in the documentation would suffice.

Community guidelines: contributions are clearly solicited but there are no guidelines as such just a very brief "Patches and/or suggestions are welcome." It would be good to have a more complete set of guidelines (they don't need to be long, but more than one sentence).

Automated tests: given that this is GUI software, I was pleased to see some automated tests :)

Documentation: a general comment on the documentation is that it would be good to have it as HTML as well as a PDF for accessibility reasons. (This isn't a requirement for JOSS as far as I know, just my personal opinion so feel free to ignore.)

I've attached some comments on the paper; nothing major, though I wasn't very keen on the parameter search section since it makes it sound like a novel algorithm whereas it's really just an exhaustive search (with some bounding applied when it is clear that the current minimum is exceeded).
[Flint.10.21105.joss.02331.pdf](https://github.com/openjournals/joss-reviews/files/5075623/Flint.10.21105.joss.02331.pdf)

Overall, this is a nice piece of software that's easy to use. 👍

**Comment by tabe** [comment]
Dear @dawbarton,

Thank you very much for your constructive comments.

> Installation: I like the list of dependencies for different Linux distributions. The one dependency that isn't listed is gnuplot - it might be worth adding that to the list given that it is required for viewing the plots. I built the package from source under Linux using the supplied Makefile and it was a smooth process.
>
Added sentences about requiring gnuplot in INSTALL.org and PORTABILITY.org.

> A statement of need: I can't see any sort of statement of need, comparison with other software, or intended audience in the documentation. There is a clear description of what the software does though. (And I can infer from that who the intended audience is, but it would be good to express it more clearly.) Simply taking some of the text from the JOSS paper and putting in an appropriate place in the documentation would suffice.
>
Improved the introductory part in README and the user guide.

> Community guidelines: contributions are clearly solicited but there are no guidelines as such just a very brief "Patches and/or suggestions are welcome." It would be good to have a more complete set of guidelines (they don't need to be long, but more than one sentence).
>
We will publish a community guidelines soon.  I've filed an issue to work: https://github.com/flintproject/Flint/issues/21

> Automated tests: given that this is GUI software, I was pleased to see some automated tests :)
>
So far we've wondered automated tests through wxWidgets' wxUIActionSimulator, but there is no specific roadmap yet.  Filed as a to-do: https://github.com/flintproject/Flint/issues/20

> Documentation: a general comment on the documentation is that it would be good to have it as HTML as well as a PDF for accessibility reasons. (This isn't a requirement for JOSS as far as I know, just my personal opinion so feel free to ignore.)
>
Agreed, we will publish the user guide in HTML at Flint's site.  Filed as another to-do: https://github.com/flintproject/Flint/issues/19.

> I've attached some comments on the paper; nothing major, though I wasn't very keen on the parameter search section since it makes it sound like a novel algorithm whereas it's really just an exhaustive search (with some bounding applied when it is clear that the current minimum is exceeded).
> [Flint.10.21105.joss.02331.pdf](https://github.com/openjournals/joss-reviews/files/5075623/Flint.10.21105.joss.02331.pdf)
>
Updated the draft according to your comments, which were all reasonable.

In Algorithm 1, we say "t found in D" instead of "t \in D" because D is not just a set of time points, but an ordered set of pairs <t, v> where v is the value at time t.
While admitting that Algorithm 1 is a simple extention of an exhaustive search, we keep it exposited in our paper for a couple of reasons: (1) it helps understand the timing (in the sense of simulation time) when each thread breaks prematurely, and (2) it clarifies that the necessary resource to be shared in parallel execution is only a double floating-point number with its mutex, which means the overhead is marginal.  If an analogy is allowed, it's like  Knuth's Algorithm T for a sequential search.  It's just a sequential search, but still worth mentioning.  Anyway it would be nice to find and cite a reference analyzing the same bounding technique with Algorithm 1 if any.

---
@whedon generate pdf

**Comment by mstimberg** [comment]
@tabe Thank you for all the changes to the instructions, user manual and article, I think they improved a lot. I agree with @dawbarton that there should be some minimal form of community guidelines (where to submit bugs, etc.), and this is the only real blocker to acceptance from my side.

**Comment by majensen** [comment]
Thanks @dawbarton and @mstimberg for all your work on this submission. 

@funasoul - I'm assuming that you are not able to review. I will unassign you at this time. If there are any issues with this, please contact me directly at maj \-dot\- fortinbras \-at\- gmail \-dot\- com.

**Comment by majensen** [comment]
@whedon remove @funasoul as reviewer

**Comment by whedon** [comment]
OK, @funasoul is no longer a reviewer

**Comment by dawbarton** [comment]
Everything is good from my perspective, I've checked off all the items on the checklist. I've made a few comments on the community guidelines in https://github.com/flintproject/Flint/commit/9029aecc7433040f405c5db501b2f6686335f150 but they are pretty minor.

**Comment by mstimberg** [comment]
Everything is good from my side as well, happy to recommend the paper's acceptance in its current form. A very minor final comment on the `CONTRIBUTING.md` document: In the end it says "Optionally, subscribe the official mailing list flint-discuss@googlegroups.com", but it's not entirely clear *how* to subscribe (send "subscribe" to that address?). Maybe it could link instead (or additionally) to the google groups website: https://groups.google.com/g/flint-discuss

**Comment by tabe** [comment]
> Maybe it could link instead (or additionally) to the google groups website: https://groups.google.com/g/flint-discuss

All right, done at https://github.com/flintproject/Flint/commit/74cd69a5b6f5abde578901240ac2cc7519971b89
Thanks @mstimberg

**Comment by majensen** [comment]
Wonderful all, I will do a light proofread of the paper and will get back to you @tabe within a day or two

**Comment by majensen** [comment]
@whedon check references

**Comment by majensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2331 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2331 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-132474d2711b/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-132474d2711b/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-132474d2711b/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by majensen** [comment]
That's weird (@arfon), didn't compile (in https://github.com/openjournals/joss-reviews/issues/2331#issuecomment-686473971)
Was able to create proof using https://whedon.theoj.org/preview, pdf attached.

[wyc0zmno62jsio38fcxy.pdf](https://github.com/openjournals/joss-reviews/files/5168663/wyc0zmno62jsio38fcxy.pdf)

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by arfon** [comment]
Sometimes Whedon is a little flaky...

**Comment by majensen** [comment]
@whedon check references

**Comment by majensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fphys.2015.00251 is OK
- 10.14326/abe.3.50 is OK
- 10.1109/SAINT.2012.40 is OK
- 10.1016/j.pbiomolbio.2004.01.004 is OK
- 10.1093/bioinformatics/btg015 is OK
- 10.3389/fphys.2010.00164 is OK
- 10.1093/bioinformatics/btl485 is OK
- 10.1126/science.290.5495.1358 is OK
- 10.1073/pnas.0406841102 is OK
- 10.1016/j.imr.2015.12.004 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by majensen** [comment]
@tabe - I am going to make one more editing pass, to get the language sounding great. Expect only one more pull request. Thanks!

**Comment by majensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by majensen** [comment]
@tabe we are ready to move forward. Can I ask you to create an archive of the Flint repository (using [Zenodo](https://zenodo.org), [FigShare](https://figshare.com), or similar)? The title of your archive should match the title of the paper. Then please report back the DOI of the archive in this thread. 
Please let me know if you have any questions -

**Comment by tabe** [comment]
@majensen, uploaded an archive to Zenodo:
https://doi.org/10.5281/zenodo.4017040
I hope this works for you.

**Comment by majensen** [comment]
@whedon set Flint-2.4.0RC1-JOSS  as version

**Comment by whedon** [comment]
OK. Flint-2.4.0RC1-JOSS  is the version.

**Comment by majensen** [comment]
@whedon set 10.5281/zenodo.4017040 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4017040" target="_blank">10.5281/zenodo.4017040</a> is the archive.

**Comment by majensen** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fphys.2015.00251 is OK
- 10.14326/abe.3.50 is OK
- 10.1109/SAINT.2012.40 is OK
- 10.1016/j.pbiomolbio.2004.01.004 is OK
- 10.1093/bioinformatics/btg015 is OK
- 10.3389/fphys.2010.00164 is OK
- 10.1093/bioinformatics/btl485 is OK
- 10.1126/science.290.5495.1358 is OK
- 10.1073/pnas.0406841102 is OK
- 10.1016/j.imr.2015.12.004 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1713

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1713, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @tabe - this looks good, and I'm working on the final processing.
Can you merge https://github.com/flintproject/Flint/pull/24, which has some minor changes to the paper?
Also, I think the SUNDIALS TOMS paper should have 10.1145/1089014.1089020 added as the DOI.

**Comment by tabe** [comment]
Dear @danielskatz,

> 👋 @tabe - this looks good, and I'm working on the final processing.
> Can you merge [flintproject/Flint#24](https://github.com/flintproject/Flint/pull/24), which has some minor changes to the paper?

Done.

> Also, I think the SUNDIALS TOMS paper should have 10.1145/1089014.1089020 added as the DOI.

Added at https://github.com/flintproject/Flint/commit/c90b99a53b3930b1e46d570e79e3e6f3d39c9b6e.

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02331/joss.02331/10.21105.joss.02331.pdf)

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3389/fphys.2015.00251 is OK
- 10.14326/abe.3.50 is OK
- 10.1109/SAINT.2012.40 is OK
- 10.1137/S0036144500378302 is OK
- 10.1016/j.pbiomolbio.2004.01.004 is OK
- 10.1093/bioinformatics/btg015 is OK
- 10.1145/1089014.1089020 is OK
- 10.3389/fphys.2010.00164 is OK
- 10.1093/bioinformatics/btl485 is OK
- 10.1126/science.290.5495.1358 is OK
- 10.1073/pnas.0406841102 is OK
- 10.1016/j.imr.2015.12.004 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1717

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1717, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1303118585074774016) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1718
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02331](https://doi.org/10.21105/joss.02331)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @mstimberg, @dawbarton for reviewing, and @majensen for editing!

And congratulations to @tabe (Takeshi Abe) and co-author!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02331/status.svg)](https://doi.org/10.21105/joss.02331)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02331">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02331/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02331/status.svg
   :target: https://doi.org/10.21105/joss.02331
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02331/status.svg)](https://doi.org/10.21105/joss.02331)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
